client/src/components/SignUpModal.jsx

```diff
@@ -1,186 +1,276 @@
+        </Form.Item>
+
+        <Form.Item
+          label="Password"
+          name="password"
+          rules={[
+            {
+              required: true,
+              message: 'Please input your password!',
+            },
+          ]}
+        >
+          <Input.Password />
+        </Form.Item>
+
+        <Form.Item
+          label="First Name"
+          name="name"
+          rules={[
+            {
+              required: true,
+              message: 'Please input your first name!',
+            },
+          ]}
+        >
+          <Input />
+        </Form.Item>
+
+        <Form.Item
+          label="Location"
+          name="location"
+          rules={[
+            {
+              required: true,
+              message: 'Please input your location!',
+            },
+          ]}
+        >
+          <Input />
+        </Form.Item>
+
+        <Form.Item
+          label="Pronouns"
+          name="pronouns"
+          rules={[
+            {
+              required: true,
+              message: 'Please select your pronouns!',
+            },
+          ]}
+        >
+          <Select placeholder="Select your pronouns"
+          options={pronounOptions} 
+          dropdownStyle={{ maxWidth: '100%', whiteSpace: 'normal' }}>
+          </Select>
+        </Form.Item>
+
+        <Form.Item
+          label="Age"
+          name="age"
+          rules={[
+            {
+              required: true,
+              message: 'Please input your age!',
+            },
+          ]}
+        >
+          <InputNumber />
+        </Form.Item>
+
+        <Form.Item
+          label="Partner?"
+          name="has_partner"
+          rules={[
+            {
+              required: true,
+              message: 'Please indicate if you have a partner!',
+            },
+          ]}
+        >
+          <Radio.Group onChange={handlePartnerChange}>
+            <Radio value={true}>Yes</Radio>
+            <Radio value={false}>No</Radio>
+          </Radio.Group>
+        </Form.Item>
+
+        {hasPartner && (
+          <>
+            <Form.Item
+              label="Partner's First Name"
+              name="partner_name"
+              rules={[
+                {
+                  required: true,
+                  message: `Please input your partner's first name!`,
+                },
+              ]}
+            >
+              <Input />
+            </Form.Item>
+
+            <Form.Item
+              label="Partner's Pronouns"
+              name="partner_pronouns"
+              rules={[
+                {
+                  required: true,
+                  message: `Please select your partner's pronouns!`,
+                },
+              ]}
+            >
+              <Select placeholder="Select your partner's pronouns"
+               options={pronounOptions} 
+               dropdownStyle={{ maxWidth: '100%', whiteSpace: 'normal' }}>
+              </Select>
+            </Form.Item>
```

Comment:
I noticed that the pronoun options were in a drop-down style, and that pronouns are displayed with the user's name in the UserProfile component. If the user chooses "other", what would be displayed with the user's name? Could there be an option for the user to input their pronouns when choosing "other"?